ci: auto-deploy website on push to master

machi1990 opened the pull request:
This is based on https://v2.docusaurus.io/docs/deployment/#triggering-deployment-with-github-actions

Fixes https://github.com/aerogear/graphback/issues/1945

craicoverflow commented:
I see the entire CI suite runs for this PR even though all that is changed is .md and .yaml

It would be amazing if what runs is based on the files changed - obviously not in this PR it is a future improvement that could prevent us hitting API limits and speed up PRs.

machi1990 commented:
> I see the entire CI suite runs for this PR even though all that is changed is .md and .yaml
> 
> It would be amazing if what runs is based on the files changed - obviously not in this PR it is a future improvement that could prevent us hitting API limits and speed up PRs.

So the entire CI is ran because I modified the two workflows. Normally, I've added "paths filtering rules" to prevent running all workflows but some of them based on changes.
